Adds Rebate and Fee Rates to Shortable Provider

AlexCatarino opened the pull request:
#### Description
Adds rebate and fee rates to the shortable provider.
We don't model the impact of these rates when the algorithm holds a short position. At this time, algorithms can fetch this information and apply it to its logic or display this information.

#### Related Issue
Closed #7614 

#### Motivation and Context
New feature. IB provides this data, so we will make it available. We don't model its impact at this time.

#### Requires Documentation Change
Yes. Update current documentation on shortable providers.

#### How Has This Been Tested?
New unit tests and enhance regression algorithms.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. <!--- If not applicable, please explain why -->
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

AlexCatarino committed:
Adds Rebate and Fee Rates to Shortable Provider

If the provider does have this information, the rates are zero.

Martin-Molinero commented on Common/Data/Shortable/LocalDiskShortableProvider.cs:
```diff
@@ -124,10 +158,12 @@ private void ClearCache()
             });
         }
+
+        protected record ShortableData(long? ShortableQuantity, decimal RebateFee, decimal FeeRate);
```
Cool!

Martin-Molinero reviewed (commented):
Nice! 
We have `bool Shortable(Symbol symbol)` in the algorithm API, should be add similar helper methods for the rebate and fee rate?

AlexCatarino committed:
Adds Helper Methods to QCAlgorithm

- Improve summaries.
- Express values as rates instead of percentages.
- Adds headers to files to emulate real data

AlexCatarino committed:
Clarify the Data Format

Martin-Molinero reviewed (approved):
Sweet!

AlexCatarino committed:
Removes Helper Methods from QCAlgorithm

AlexCatarino committed:
Removes `Shortable` and `ShortableQuantity` from Example

Algorithms should prefer getting the information from the `Security.ShortableProvider`.
